app/models/position_history.rb

```diff
@@ -3,5 +3,39 @@ class PositionHistory < ApplicationRecord
   belongs_to :position
 
   validates :started_on, presence: true
-  validates :finished_on, presence: true
+  validate :last_position_finished, on: :create
+  validate :validate_position_history_overlap
+  validate :manager_exists
+
+  def last_position_finished
+    employee = Employee.find_by_id(employee_id)
+    position = employee.position_histories.last
+    if position.present? && position.finished_on.nil?
```

Comment:
`if PositionHistory.where(employee_id: employee_id).order(id: :asc).take(1).finished_on.nil?`

This will make only 1 query to db.